### DISCUSSION TITLE
Unexpected base commit in reply to GET pull request

### DISCUSSION THREAD

**Discussion by Xavier-explore-dev** [community/community#24563] [Apps, API and Webhooks]
<p>Hello,</p>
<p>I’m trying to read the base and head commits of the following public pull request:</p>
<p><a href="https://github.com/explore-dev/dogfood-ts/pull/3" rel="nofollow noopener">https://github.com/explore-dev/dogfood-ts/pull/3</a> </p>
<p>This pull request merges only one commit from branch <em><a href="https://github.com/explore-dev/dogfood-ts/tree/feat-add-branch-names" title="explore-dev/dogfood-ts:feat-add-branch-names" rel="nofollow noopener">feat-add-branch-name</a></em> to <em>master</em>. The commit hash is:</p>
<pre><code>84a7c9feb0f5067906a8433842884cf1a851e5dd
</code></pre>
<p>The parent commit from master is:</p>
<pre><code>8f02d01cb137212fc7159de8e0c1324f686a118a
</code></pre>
<p>To get information about the pull request, I’m performing the following request:</p>
<pre><code>curl https://api.github.com/repos/explore-dev/dogfood-ts/pulls/3
</code></pre>
<p>The reply gives the following base and head commits:</p>
<pre><code>Head: 84a7c9feb0f5067906a8433842884cf1a851e5dd
Base: 56e32028da4e0a9e8745ece9c634bb3ec380ffbe
</code></pre>
<p>The head commit is correct. However, the base commit is not the parent of the head commit, but a more recent commit in master instead. I know that the pull request has conflicts, so I imagine github already created a merge commit from master. However, this merge commit should not be the head commit of the branch: the conflicts were not resolved. Is this a problem or am I missing something.</p>
<p>Thank you!</p>
<p>Regards,</p>
<p>Xavier</p>

**Comment by lesleyanneb** [comment, marked as answer]
<p>Hi there,</p>
<p>Thank you for contributing to the GitHub community forum!</p>
<p>Taking a look at <a href="https://github.com/explore-dev/demo-ts/pull/3" rel="nofollow noopener">explore-dev/demo-ts pull request 3</a> it appears that there were subsequent force pushes on the 28th after 56e32028da4e0a9e8745ece9c634bb3ec380ffbe<br>
was pushed to master which could be the reason the pr base ref is now set to commit SHA<br>
56e32028da4e0a9e8745ece9c634bb3ec380ffbe.</p>
<p>I hope this helps to explain. Please let us know if you need anything else!</p>
